build_tools/setup_helpers/extension.py

```diff
@@ -116,27 +120,86 @@ def _build_third_party():
     )
 
 
+def _get_ext(debug):
+    return CppExtension(
+        _EXT_NAME,
+        _get_srcs(),
+        libraries=_get_libraries(),
+        include_dirs=_get_include_dirs(),
+        extra_compile_args=_get_eca(debug),
+        extra_objects=_get_extra_objects(),
+        extra_link_args=_get_ela(debug),
+    )
+
+
+def _get_ext_transducer(debug):
+    base_path = _TP_TRANSDUCER_BASE_DIR
+    default_warp_rnnt_path = base_path / "build"
+
+    include_dirs = [os.path.realpath(os.path.join(base_path, 'include'))]
+
+    if "WARP_RNNT_PATH" in os.environ:
+        warp_rnnt_path = os.environ["WARP_RNNT_PATH"]
+    else:
+        warp_rnnt_path = default_warp_rnnt_path
+
+    librairies = ['warprnnt']
+    if platform.system() == 'Darwin':
+        lib_ext = ".dylib"
+    else:
+        lib_ext = ".so"
+    extra_objects = [str(os.path.join(warp_rnnt_path, f'lib{l}{lib_ext}')) for l in librairies]
+
+    extra_compile_args = ['-fPIC']
+    extra_compile_args += ['-std=c++14']
+
+    # TODO Enable GPU compilation
+    if False and torch.cuda.is_available():
+
+        if "CUDA_HOME" not in os.environ:
+            raise RuntimeError("Please specify the environment variable CUDA_HOME.")
+
+        extra_compile_args += ['-DWARPRNNT_ENABLE_GPU']
+
+    else:
+        print("Not building GPU extensions.")
+
+    # if platform.system() == 'Darwin':
+    #     root_dir = "@loader_path"
+    # else:
+    #     root_dir = "$ORIGIN"
+    # rel_warp_rnnt_path = os.path.join(root_dir, 'third_party', 'warp_transducer', 'build')
+    rel_warp_rnnt_path = os.path.realpath(warp_rnnt_path)
+    print("path for object:", rel_warp_rnnt_path)
+    return CppExtension(
+        name='_warp_transducer',
+        sources=[os.path.realpath(base_path / 'pytorch_binding' / 'src' / 'binding.cpp')],
+        # sources=[os.path.realpath(_TP_BASE_DIR / 'binding.cpp')],
+        include_dirs=include_dirs,
+        extra_objects=extra_objects,
+        library_dirs=[os.path.realpath(warp_rnnt_path)],
+        libraries=librairies,
+        extra_link_args=['-Wl,-rpath,' + rel_warp_rnnt_path],
```

Comment:
The linux/macos builds are not finding the shared library, see [here](https://app.circleci.com/pipelines/github/pytorch/audio/4185/workflows/53e9a945-8248-4283-a0f9-290f10ebd0fa/jobs/129878
):
```
OSError: libwarprnnt.so: cannot open shared object file: No such file or directory
```
though I do find it (libwarprnnt.so) when sshing in the machine:
```
./opt/conda/conda-bld/torchaudio_1608332828661/work_moved_torchaudio-0.8.0.dev20201218-py38_linux-64/third_party/warp_transducer/build/libwarprnnt.so
```
The suggestion I saw does say to specify `rpath`, see [doc](https://docs.conda.io/projects/conda-build/en/latest/resources/use-shared-libraries.html) or [here](https://github.com/conda-forge/cmake-feedstock/issues/21#issuecomment-256090118). I haven't had success with this though. Thoughts?

hey, quick question: i'm trying to build a second C++ extension for torchaudio, and I'm running into an issue linking for conda, but not for the rest. when i ssh in the machine, i find the library (libwarprnnt.so). https://app.circleci.com/pipelines/github/pytorch/audio/4185/workflows/53e9a945-8248-4283-a0f9-290f10ebd0fa/jobs/129878 -- thoughts?

Reply:
cc @malfet for this as well

Reply:
Are you sure the binary is actually in the right location upon creation of the conda package? That is, is libwarprnnt.so being moved / included in the move into the folder used to build the conda package?

Reply:
So the `libwarprnnt.so` is found in `third_party/warp_transducer/submodule/build/libwarprnnt.so`.

This explains why `conda` package cannot find it. Because this is not part of the package data.
Currently in torchaudio, when binding libsox, we do static linking, meaning we build libsox then statically link it to C++ extension (_torchaudio.so). So this problem does not happen.

The solution is either
1. link libwarprnnt statically 
2. think of a way (and a convention to be followed in the future) where to put dynamic library and how to add it to library look up path at runtime.

Reply:
If `libwarprnnt` allows static build, then it's easy to resolve

Reply:
yeah, I'll just build statically